advanced_alchemy/service/typing.py

```diff
@@ -20,18 +21,29 @@
     runtime_checkable,
 )
 
-from typing_extensions import TypeAlias, TypeGuard
+from typing_extensions import Annotated, TypeAlias, TypeGuard
 
 from advanced_alchemy.filters import StatementFilter  # noqa: TCH001
 from advanced_alchemy.repository.typing import ModelT
 
+T = TypeVar("T")  # pragma: nocover
 try:
     from pydantic import BaseModel  # pyright: ignore[reportAssignmentType]
+    from pydantic.config import ConfigDict  # pyright: ignore[reportAssignmentType]
     from pydantic.type_adapter import TypeAdapter  # pyright: ignore[reportUnusedImport, reportAssignmentType]
 
     PYDANTIC_INSTALLED: Final[bool] = True
 except ImportError:  # pragma: nocover
 
+    class ConfigDict:  # type: ignore[no-redef] # pragma: nocover
```

Comment:
could you explain for my benefit the need for these?

Reply:
It's to make sure the type hinting is happy in the `_util.py` and other places within the service layer.  Since it's designed to work with or without Pydantic and Msgspec, it was much easier to centralize all of that conditional logic here than have it spread over several files.

Reply:
We actually don't need `ConfigDict` though.  It was or a tweak I removed.